ddtrace/debugging/_signal/context.py

```diff
@@ -0,0 +1,95 @@
+from collections import ChainMap
+import typing as t
+
+from ddtrace.debugging._probe.model import ProbeConditionMixin
+from ddtrace.debugging._probe.model import ProbeEvalTiming
+from ddtrace.debugging._probe.model import RateLimitMixin
+from ddtrace.debugging._probe.model import TimingMixin
+from ddtrace.debugging._signal.model import Signal
+from ddtrace.debugging._signal.model import SignalState
+from ddtrace.internal.compat import ExcInfoType
+from ddtrace.internal.rate_limiter import RateLimitExceeded
+
+
+class SignalContext:
+    """Debugger signal context manager for function probes.
+
+    Probes with an evaluation timing will trigger the following logic:
+
+    - If the timing is on ENTRY, the probe will trigger on entry and on exit,
+      but the condition and rate limiter will only be evaluated on entry.
+
+    - If the timing is on EXIT, the probe will trigger on exit only, and the
+      condition and rate limiter will be evaluated on exit.
+
+    - IF the timing is DEFAULT, the probe will trigger on the default timing for
+      the signal.
+
+    - If the probe has no timing, it defaults to the ENTRY timing.
+    """
+
+    def __init__(self, signal: Signal) -> None:
+        self.signal = signal
+        self._timing = ProbeEvalTiming.ENTRY
+
+        probe = signal.probe
+        if isinstance(probe, TimingMixin):
```

Comment:
nit: would easier to read if there is else under this if with `self._timing = ProbeEvalTiming.ENTRY`